Runtime/AvatarCreator/Scripts/UI/Elements/SignupElement.cs

```diff
@@ -34,15 +36,25 @@ private void OnDisable()
             continueWithoutSignupButton.onClick.RemoveListener(OnContinueWithoutSignupButton);
         }
 
+        private void OnDestroy()
+        {
+            cancellationTokenSource.Cancel();
+            cancellationTokenSource.Dispose();
+        }
+
         private void OnEmailChanged(string newEmailValue)
         {
             sendEmailButton.interactable = !string.IsNullOrEmpty(newEmailValue) && ValidatorUtil.IsValidEmail(newEmailValue);
         }
 
-        private void OnSendEmailButton()
+        private async void OnSendEmailButton()
         {
             var email = emailField.text;
-            AuthManager.Signup(email);
+            await TaskExtensions.HandleCancellation(AuthManager.Signup(email, cancellationTokenSource.Token), () => OnEmailSent(email));
```

Comment:
same as before I'm not really sure why this HandleCancellation function is needed. The cancellation and exception throwing should be in the AuthManager.Signup or AuthAPIRequests.Signup. that way it will automatically be available for anybody using our public API's too

Reply:
It's not throwing an exception, but catching it here so that it wouldn't fall to the next level and throw unhandled exceptions. Elements are the last layer, that should handle the exceptions, that is not called out from next level (Creator controllers, that create buttons, etc..)